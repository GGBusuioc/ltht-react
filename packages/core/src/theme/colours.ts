export const PRIMARY_TEXT_COLOUR = '#000000'
export const SECONDARY_TEXT_COLOUR = '#6E777A'
export const DANGER_TEXT_COLOUR = '#'
export const INFO_TEXT_COLOUR = '#'

export const ICON_RED_COLOUR = '#'
export const ICON_AMBER_COLOUR = '#'
export const ICON_GREEN_COLOUR = '#'
