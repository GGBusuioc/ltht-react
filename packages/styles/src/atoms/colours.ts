import * as colours from './nhs-colours'

export const TEXT_PRIMARY_COLOUR = colours.NHS_BLACK
export const TEXT_SECONDARY_COLOUR = colours.NHS_MID_GREY
export const TEXT_DANGER_COLOUR = colours.NHS_DARK_RED
export const TEXT_INFO_COLOUR = colours.NHS_DARK_BLUE

export const ICON_PRIMARY_COLOUR = colours.NHS_BLACK
export const ICON_DANGER_COLOUR = colours.NHS_EMERGENCY_SERVICES_RED
export const ICON_WARNING_COLOUR = colours.NHS_ORANGE
export const ICON_SUCCESS_COLOUR = colours.NHS_GREEN
export const ICON_DEFAULT_COLOUR = colours.NHS_MID_GREY

export const CARD_BACKGROUND_COLOUR = colours.NHS_WHITE
export const CARD_BACKGROUND_COLOUR_NO_DATA = '#F2F2F2'
export const CARD_SHADOW_COLOUR = '#666666'
export const CARD_LIST_ITEM_BACKGROUND_HOVER = colours.NHS_PALE_GREY
export const CARD_LIST_ITEM_BORDER_HOVER = '#b0b0b0'
