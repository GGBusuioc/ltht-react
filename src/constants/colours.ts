export const BUTTON_BLUE = '#0066CC'
export const BUTTON_GREEN = '#006B54'
