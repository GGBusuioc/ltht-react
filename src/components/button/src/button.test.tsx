import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { matchers } from 'jest-emotion'

import Button from './button.fixtures'
import { BUTTON_BLUE, BUTTON_GREEN } from '../../../constants/colours'

expect.extend(matchers)

describe('button', () => {
  describe('basics', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div')
      ReactDOM.render(<Button>Example</Button>, div)
    })
    it('matches snapshot', () => {
      const wrapper = renderer.create(<Button>Primary Button</Button>).toJSON()
      expect(wrapper).toMatchSnapshot()
    })
  })
})

describe('button', () => {
  describe('styles', () => {
    it('default', () => {
      const wrapper = renderer.create(<Button>Default Button</Button>).toJSON()
      expect(wrapper).toMatchSnapshot()
      expect(wrapper).toHaveStyleRule('background', 'transparent')
      expect(wrapper).toHaveStyleRule('color', BUTTON_BLUE)
      expect(wrapper).toHaveStyleRule('background', '#e5e5e5', { target: ':hover' })
    })
    it('primary', () => {
      const wrapper = renderer.create(<Button primary>Primary Button</Button>).toJSON()
      expect(wrapper).toMatchSnapshot()
      expect(wrapper).toHaveStyleRule('background', BUTTON_BLUE)
      expect(wrapper).toHaveStyleRule('color', 'white')
      expect(wrapper).toHaveStyleRule('background', '#005bb7', { target: ':hover' })
    })
    it('workflow', () => {
      const wrapper = renderer.create(<Button workflow>Workflow Button</Button>).toJSON()
      expect(wrapper).toMatchSnapshot()
      expect(wrapper).toHaveStyleRule('background', BUTTON_GREEN)
      expect(wrapper).toHaveStyleRule('color', 'white')
      expect(wrapper).toHaveStyleRule('background', '#00604b', { target: ':hover' })
    })
  })
})
