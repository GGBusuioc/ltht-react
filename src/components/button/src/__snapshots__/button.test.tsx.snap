// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button basics matches snapshot 1`] = `
.emotion-0 {
  background: transparent;
  box-shadow: none;
  color: #0066CC;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  display: block;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.2;
  margin: 2px 2px;
  padding: 1px 7px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
}

.emotion-0:hover {
  background: #e5e5e5;
}

<button
  className="emotion-0"
>
  Primary Button
</button>
`;

exports[`button styles default 1`] = `
.emotion-0 {
  background: transparent;
  box-shadow: none;
  color: #0066CC;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  display: block;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.2;
  margin: 2px 2px;
  padding: 1px 7px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
}

.emotion-0:hover {
  background: #e5e5e5;
}

<button
  className="emotion-0"
>
  Default Button
</button>
`;

exports[`button styles primary 1`] = `
.emotion-0 {
  background: transparent;
  box-shadow: none;
  color: #0066CC;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  display: block;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.2;
  margin: 2px 2px;
  padding: 1px 7px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
  background: #0066CC;
  color: white;
}

.emotion-0:hover {
  background: #e5e5e5;
}

.emotion-0:hover {
  background: #005bb7;
}

<button
  className="emotion-0"
>
  Primary Button
</button>
`;

exports[`button styles workflow 1`] = `
.emotion-0 {
  background: transparent;
  box-shadow: none;
  color: #0066CC;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  display: block;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.2;
  margin: 2px 2px;
  padding: 1px 7px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
  background: #006B54;
  color: white;
}

.emotion-0:hover {
  background: #e5e5e5;
}

.emotion-0:hover: {
  background: #00604b;
}

<button
  className="emotion-0"
>
  Workflow Button
</button>
`;
